From: =?utf-8?q?Michel_D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Wed, 2 Apr 2025 09:45:55 +0200
Subject: cursor-sprite/xcursor: Reset current_frame only if xcursor_images
 change

Otherwise current_frame always stays at 0, and animated cursors don't
actually animate.

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/4030
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2106097
Origin: Upstream commit 56a583fe346ac4eed12075fca65e95c6713aa84a in 48.2
---
 src/backends/meta-cursor-sprite-xcursor.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/backends/meta-cursor-sprite-xcursor.c b/src/backends/meta-cursor-sprite-xcursor.c
index ac34736..a2dad50 100644
--- a/src/backends/meta-cursor-sprite-xcursor.c
+++ b/src/backends/meta-cursor-sprite-xcursor.c
@@ -500,12 +500,11 @@ load_cursor_from_theme (MetaCursorSprite *sprite)
                            xcursor_images);
     }
 
-  sprite_xcursor->current_frame = 0;
-
   if (sprite_xcursor->xcursor_images == xcursor_images)
     return FALSE;
 
   sprite_xcursor->xcursor_images = xcursor_images;
+  sprite_xcursor->current_frame = 0;
   load_from_current_xcursor_image (sprite_xcursor);
   return TRUE;
 }
